sentry-android-core/src/main/java/io/sentry/android/core/Installation.java

```diff
@@ -0,0 +1,45 @@
+package io.sentry.android.core;
+
+import android.content.Context;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.charset.Charset;
+import java.util.UUID;
+
+final class Installation {
+  private static String sID = null;
+  private static final String INSTALLATION = "INSTALLATION";
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  private Installation() {}
+
+  public static synchronized String id(Context context) {
+    if (sID == null) {
+      File installation = new File(context.getFilesDir(), INSTALLATION);
+      try {
+        if (!installation.exists()) writeInstallationFile(installation);
+        sID = readInstallationFile(installation);
```

Comment:
The id can be returned from the `writeInstallationFile` instead of doing I/O here

Reply:
good catch